osh/sh_expr_eval.py

```diff
@@ -1042,14 +1046,17 @@ def EvalB(self, node):
                     raise AssertionError(op_id)  # should never happen
 
                 if arg_type == bool_arg_type_e.Str:
+                    eval_flags = 0
+                    if self.exec_opts.nocasematch():
+                        eval_flags |= FnmCasefold
```

Comment:
how about calling these fnmatch_flags and regex_flags ?

Also in this case we can probably use a ternary expression

```
fnmatch_flags = FNM_CASEFOLD if self.exec_opts.nocasematch() else 0
```